src/components/CytoscapeLayout/CytoscapeLayout.tsx

```diff
@@ -118,13 +97,47 @@ export default class CytoscapeLayout extends React.Component<CytoscapeLayoutProp
         this.handleMouseOut(cytoscapeEvent);
       }
     });
+
+    // when the graph is fully populated and ready, we need to perform additional things
     this.cy.ready((evt: any) => {
       if (!this.props.isReady) {
+        // Don't allow a large zoom if the graph has a few nodes (nodes would look too big).
+        if (this.cy.zoom() > 2.5) {
+          this.cy.zoom(2.5);
+          this.cy.center();
+        }
+
+        this.processGraphUpdate();
         this.props.onReady(evt.cy);
       }
     });
   }
 
+  // Perform additional processing on the graph to do things like decorating it with badges
+  processGraphUpdate() {
+    console.log('CY: graph was updated and needs to be processed further');
+
+    // TODO: refactor the GraphBadge.XXX classes so the constructor doesn't take an
+    //       element. Pass ele to the buildBadge and destroyBadge funcs so we don't have to
+    //       create so many objects - we should just create one object per badge type
+    //       and let it build/destroy multiple badges.
+    this.cy.startBatch();
+    this.cy.nodes().forEach(ele => {
+      if (!this.props.badgeStatus.hideCBs && ele.data('hasCB') === 'true') {
```

Comment:
Truthy makes `=== 'true'` redundant.

Reply:
this code is going to change in the near future (see the TODO a couple lines above), but ... duly noted.